stock_override_procurement/models/procurement_group.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProcurementGroup(models.Model):
+
+    _inherit = "procurement.group"
+
+    @api.model
+    def run(self, procurements, raise_user_error=True):
+        procurements = self._override_procurements(procurements)
+        return super().run(procurements, raise_user_error=raise_user_error)
+
+    @api.model
+    def _override_procurements(self, procurements):
+        return [
+            self._get_override_procurement(procurement) for procurement in procurements
+        ]
+
+    @api.model
+    def _get_override_procurement(self, procurement):
+        values = procurement.values
+
+        def _get_override_value(field_name):
+            return (
+                values.get(field_name)
+                if values.get(field_name)
```

Comment:
```suggestion
                if field_name in values
```
required to allows to unset a value by override